### ISSUE TITLE
Thunderbird v128.0.1

### ISSUE THREAD

**Issue by eddieturtle** [xnotepp/xnote#140]
XNote++ is now broken. Any updates coming to fix this?

**Comment by opto** [comment]
yes, it is being worked on. TB introduced a new module system, and now there are timing problems, so that it may start, or nt at TB restart.

**Comment by Chabi01** [comment]
Thanks a lot to work on this update : I'm using Xnote all the time !
Thanks !! :)

**Comment by Chealer** [comment]
Can you please rename this to "Broken on Thunderbird 128.0.1", or―even better―an even more specific title?
Are you saying it worked with 128.0.0?

**Comment by Chabi01** [comment]
Hi, this is not my post, but on TB128.0esr 64bits, I get a "not compatible" for the module for XNote 4.1.12.

Best,
Xavier

**Comment by JohnAnderson** [comment]
In my experience, it has never worked on TB128 (I'm using 128.1.0esr now), but it would be great if it did!

**Comment by pmruzicka** [comment]
How can we try get the old version to work?

**Comment by opto** [comment]
A test version will be coming next week,

Klaus

**Comment by pmruzicka** [comment]
Thanks for the update!

**Comment by opto** [comment]
test version for TB 128 is here: https://github.com/xnotepp/xnote/releases/tag/v4.3.24

**Comment by PeterMacej** [comment]
Seems to work fine. Thanks.

**Comment by JohnAnderson** [comment]
Seems to work for me, too.  Thanks.

John E Anderson, PhD
Associate Lecturer Emeritus of Physics, UNF
Executive Director, Jacksonville Joe Berg Seminars<http://www.joeberg.org/>
+1-781-816-9164 (Google Voice)
***@***.******@***.***>
Science + Culture → Future
***@***.***
On 2024-09-08 17:09, Peter Macej wrote:

Seems to work fine. Thanks.

**Comment by opto** [comment]
a new test version is here: https://github.com/xnotepp/xnote/releases/tag/v4.3.25

**Comment by zevrix** [comment]
Thank you for keeping the support!

I wonder if the new version is supposed to work with the latest TB 128, too?

I'm trying it with TB 128.2.3 on macOS Sequoia.

XNote does work but not quite smoothly. The size in the Size column shows up briefly when selecting a different message. Also, the Size column is populated with size values once you scroll down to a previously unseen part of the message list. A message should be selected to initiate the XNote display in the column.

P.S. Really, TB developers should work with you closely to restore the full XNote support in any version of TB. In fact, this should be their highest priority. XNote is what makes TB stand out and what makes it miles above any other email client on the market. I'll post on bugzilla soon to remind them - again! - about the importance of uninterrupted XNote support.

**Comment by opto** [comment]
new version: https://github.com/xnotepp/xnote/releases/tag/4.3.27

**Comment by zevrix** [comment]
Thanks for the new version.

However, the issues I described above remain the same so far.

That is, the Size column gets populated by the size values when scrolling up and down the message list.

Also, when creating a new note it doesn't appear in the Size column until I click away from the selected message.

**Comment by opto** [comment]
some time later this year, there will come a new method to create the column which solves these problems.

Klaus

**Comment by zevrix** [comment]
> some time later this year, there will come a new method to create the column which solves these problems.

that's good news!

**Comment by opto** [comment]
a new version with the new column is in review. It works, but columnpicker does not propagate the settings to other folders. So it can be switched on/off in the addon settings

**Comment by goozak** [comment]
While there is a separate column in the new version (4.3.32), the Size column is still re-labled 'XNote / Size' and shows the XNote.
And if I uncheck the "Show XNote in the 'size' column" option, the new standalone 'XNote' column also disappears.
I don't understand the context of your comment about "can be switched on/off in the addon settings"...

**Comment by opto** [comment]
thanks. It seems I forgot to switch off the size column. I will do that in the next version. So you can disable the size column in the column picker.

And the settings are supposed to switch off the new column, because column picker worked only intermittently here.

**Comment by Chealer** [comment]
Thank you very much @opto and @goozak 
I haven't tried Thunderbird 128 yet but is there any ticket about the intermittent bug @opto just mentioned? I would like to understand it before upgrading.

**Comment by zevrix** [comment]
Ok, the separate XNote column now indeed works in 4.3.32. That's great!

There's still a bug where the newly created note doesn't appear in the column until you deselect the message (and the situation is the same when deleting a note). I'll submit it as a separate issue for clarity sake.

(There are also a couple of minor cosmetic bugs; I'll post them separately, too).

**Comment by opto** [comment]
note not appearing: that is a TB bug. Maybe select another message, that go backto the previous one?

They totally changed the code for the message list,  I di not know whether there is anofficial  refresh function

**Comment by zevrix** [comment]
> note not appearing: that is a TB bug. Maybe select another message, that go backto the previous one?
> 
> They totally changed the code for the message list, I di not know whether there is anofficial refresh function

Yes, that's what I'm doing: select another message, then re-select the desired one.

So as far as I understand, you don't have any control over it... I'll submit a bug on bugzilla then.

**Comment by zevrix** [comment]
> note not appearing: that is a TB bug. Maybe select another message, that go backto the previous one?
> 
> They totally changed the code for the message list, I di not know whether there is anofficial refresh function

I finally posted this issue on bugzilla:
https://bugzilla.mozilla.org/show_bug.cgi?id=1947080

@opto: I wonder if you can join that bug and provide some additional info?

**Comment by opto** [comment]
I just saw, they added a new function:  refreshCustomColumn

so this will be fixed soon.

Klaus

**Comment by zevrix** [comment]
> I just saw, they added a new function: refreshCustomColumn
> so this will be fixed soon.
> Klaus

That's great news! Thanks Klaus.

**Comment by opto** [comment]
I think these can be closed.

You can test for 136 here: https://github.com/xnotepp/xnote/releases/tag/v4.5.38
